web/src/pages/api/public/ingestion.ts

```diff
@@ -623,3 +593,39 @@ export const parseSingleTypedIngestionApiResponse = <T extends z.ZodTypeAny>(
   // should not fail in prod but just log an exception, see above
   return results[0].result as z.infer<T>;
 };
+
+export const forwardLegacyEventToIngestionApi = async (
+  req: NextApiRequest,
+  batch: IngestionBatchEventType,
+): Promise<{ id: string; status: number }> => {
+  if (batch.length !== 1) {
+    throw new Error("Batch must have exactly one event");
+  }
+
+  const event = batch.shift();
+  if (!event.body.id) {
+    event.body.id = randomUUID();
+  }
+
+  const protocol = req.headers.host?.includes("localhost") ? "http" : "https";
+  const result = await fetch(
+    `${protocol}://${req.headers.host}${env.NEXT_PUBLIC_BASE_PATH ?? ""}/api/public/ingestion`,
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: req.headers.authorization as string,
+      },
+      body: JSON.stringify({ batch: [event] }),
+    },
+  );
```

Comment:
@maxdeichmann Do you see any potential issues with this part here? It works locally and I'll also test it on staging. I would expect that it should behave correctly across essentially all hosting setups as the headers are set via next.